src/wp-includes/functions.wp-scripts.php

```diff
@@ -166,20 +168,23 @@ function wp_add_inline_script( $handle, $data, $position = 'after' ) {
  *                                    as a query string for cache busting purposes. If version is set to false, a version
  *                                    number is automatically added equal to current installed WordPress version.
  *                                    If set to null, no version is added.
- * @param bool             $in_footer Optional. Whether to enqueue the script before `</body>` instead of in the `<head>`.
- *                                    Default 'false'.
+ * @param array             $args     {
+ *      Optional. An array of additional script loading strategies. Default empty array.
+ *
+ *      @type boolean   $in_footer    Optional. Default false.
+ *      @type string    $strategy     Optional. Values blocking|defer|async. Default 'blocking'.
+ * }
  * @return bool Whether the script has been registered. True on success, false on failure.
  */
-function wp_register_script( $handle, $src, $deps = array(), $ver = false, $in_footer = false ) {
+function wp_register_script( $handle, $src, $deps = array(), $ver = false, $args = array() ) {
 	_wp_scripts_maybe_doing_it_wrong( __FUNCTION__, $handle );
 
 	$wp_scripts = wp_scripts();
 
 	$registered = $wp_scripts->add( $handle, $src, $deps, $ver );
-	if ( $in_footer ) {
-		$wp_scripts->add_data( $handle, 'group', 1 );
+	if ( ! empty( $args ) ) {
+		$wp_scripts->add_data( $handle, 'script_args', $args );
```

Comment:
Technically speaking, this could be considered a backward compatibility break. How about we keep using this `group` name for `$args['in_footer']` like before? We can handle `$strategy` in any way we like, but the existing parameter should probably continued to be handled in the same way.

Reply:
@felixarntz For newer logic, we do have kept the `group` logic for footer,(but) here https://github.com/WordPress/wordpress-develop/blob/425a905a46406b93d061fccdd3ac8b0b890e9eae/src/wp-includes/class-wp-scripts.php#L798

Reply:
Can we do something similar to this. https://github.com/WordPress/wordpress-develop/commit/40d834dfc71d0c93d4f7e262e0768b2aeaf977e4

Reply:
Implemented in https://github.com/10up/wordpress-develop/pull/64